int16 sx
int16 sy
int16 gx
int16 gy
int16 cx
int16 cy
